app/settings/profiles.coffee

```diff
@@ -47,6 +48,22 @@ class Profiles extends React.Component
 
     @setState(activeProfile: id)
 
+  removeProfile: (id) ->
+    settings = @settings()
+    profileId = ""
+    tempState = {}
+    Object.assign(tempState, settings)
+    if tempState.activeProfile == id
+      profileId = Object.keys(tempState.profiles)[0]
+      tempState.activeProfile = Object.keys(tempState.profiles)[0]
+
+    delete tempState.profiles[id]
+    @_configFile.write(tempState)
+
+    @setState(
+      activeProfile: profileId, profiles: Object.values(settings.profiles)
```

Comment:
Just use tempState.activeProfile and remove the temp variable profileId

Reply:
after you write the tempState is null

Reply:
stand corrected, after moving to `profiles.coffee` this was allowed. so corrected it in new commit @npezza93